atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/ConnectionConfig.java

```diff
@@ -191,4 +202,10 @@ protected void onAcquire(Connection conn) {
             }
         });
     }
+
+    @JsonIgnore
+    @Value.Default
+    public String getSqlExceptionOverrideClassname() {
```

Comment:
Since this isn't supposed to be set via config but rather hardcoded config subclasses, let's use `Optional<SqlExceptionOverride>` and get the class name when constructing the Hikari config. 

You'd pay the cost of needing to instantiate the class where you previously did not have to, but it'd prevent future changes from typos or other classes of human errors.

If that's not feasible, let's still use `Optional<String>`